core/app/models/spree/payment.rb

```diff
@@ -196,6 +196,11 @@ def store_credit?
       payment_method.try!(:store_credit?)
     end
 
+    # @return [String] the (computed) redirection url for offsite payment. returning nil for onsite payment methods
+    def redirect_url
+      payment_method.redirect_url(self)
```

Comment:
Just a minor annoyance, but I think that `Spree::Payment.new.redirect_url` would raise an error, as `payment_method` is `nil`

Reply:
that is correct, the redirect_url needs a payment method, we could use a safe operator here to just return nil instead of throwing an error though.